tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,527 @@
+            target_dataset_attributes.create(new_attr_spec)
+            LOGGER.info(f"Updated attribute '{attribute_name}' in {dataset_name}")
+        else:
+            LOGGER.info(
+                f"""The attribute '{attribute_name}' in {dataset_name} curently has
+                 the type '{str(old_type_class)}'. Set 'override_existing_types' to
+                 True to update the type to '{str(new_type_class)}'
+                """
+            )
+
+    return dataset
+
+
+def delete_attributes(*, dataset: Dataset, attributes: Iterable[str] = None,) -> Dataset:
+    """Remove attributes from dataset by attribute name
+
+    Args:
+        dataset: An existing TUC dataset
+        attributes: list of attribute names to delete from dataset
+
+    Returns:
+        Updated Dataset
+
+    Raises:
+        ValueError: If the dataset is not a source dataset
+        ValueError: If a passed attribute does not exist in the dataset
+        ValueError: If a passed attribute is a primary key and can't be removed
+        TypeError: If the attributes argument is not an Iterable
+    """
+    dataset_name = dataset.name
+    if dataset.upstream_datasets():
+        raise ValueError(f"{dataset_name} is not a source dataset")
+
+    # Check input type is correct
+    if not isinstance(attributes, Iterable):
+        raise TypeError("attributes arg must be an Iterable")
+
+    # Get current dataset attributes
+    target_dataset_attributes = dataset.attributes
+    target_attribute_dict = {}
+    for attr in target_dataset_attributes.stream():
+        target_attribute_dict[attr.name] = attr
+    existing_attributes = target_attribute_dict.keys()
+    primary_keys = dataset.spec().to_dict()["keyAttributeNames"]
+
+    # Check all attributes exist before starting to remove any
+    for attribute_name in attributes:
+        if attribute_name not in existing_attributes:
+            raise ValueError(f"The attribute '{attribute_name}' does not exist in {dataset_name}")
+        elif attribute_name in primary_keys:
+            # Can not edit a primary key
+            raise ValueError(
+                f"The attribute '{attribute_name}' is a primary key and can't be removed"
+            )
+
+    # Remove attributes from dataset
+    for attribute_name in attributes:
+        target_dataset_attributes.delete_by_resource_id(
+            target_attribute_dict[attribute_name].resource_id
+        )
+        LOGGER.info(f"Deleted attribute '{attribute_name}' in {dataset_name}")
+
+    return dataset
+
+
+def _make_spec_dict(
+    attribute_name: str,
+    attribute_types: Dict[str, attribute_type.AttributeType],
+    attribute_descriptions: Dict[str, str],
+) -> JsonDict:
+    """
+    Create attribute spec dictionary. The default attribute type will be ARRAY STRING. To
+    set non-default attribute types, you must define them in the attribute_types
+    dictionary. Any attribute descriptions can be specified in the attribute_descriptions
+    dictionary.
```

Comment:
nit: format and consistent grammar
```suggestion
    """Create attribute spec dictionary
    
    The default attribute type will be ARRAY STRING. Non-default attribute types can be specified
    in the attribute_types dictionary. Any attribute descriptions can be specified in the
    attribute_descriptions dictionary.
```